### PR TITLE
Fixed defect when AWS region was not set for core addon version look-up

### LINKED ISSUE #1015: VpcCniAddOn: As of 1.14.1, an error stating "Region is missing" is throwing in console
### Describe the bug

After the latest upgrade we're seeing ERROR messages in our console during cdk deploy regarding "Region is missing".  It seems like three add-ons are affected:  VpcCniAddOn, CoreDnsAddOn, and KubeProxyAddOn.  This issue doesn't seem to actually affect the deploy, and isn't causing problems, but it seems like a regression regardless. 

### Expected Behavior

No error is thrown by these add-ons.

### Current Behavior

Stacktrace:

```
2024-06-11 19:58:59.398	WARN	/node_modules/@aws-quickstart/eks-blueprints/lib/addons/core-addon/index.ts:208	main	
 Error  Region is missing
error stack:
  • index.js	default
	/node_modules/@smithy/config-resolver/dist-cjs/index.js:117
  • index.js	
	/node_modules/@smithy/node-config-provider/dist-cjs/index.js:90
  • index.js	
	/node_modules/@smithy/property-provider/dist-cjs/index.js:97
  • index.js	coalesceProvider
	/node_modules/@smithy/property-provider/dist-cjs/index.js:124
  • index.js	
	/node_modules/@smithy/property-provider/dist-cjs/index.js:135
  • index.js	region
	/node_modules/@smithy/config-resolver/dist-cjs/index.js:142
  • httpAuthSchemeProvider.js	Object.defaultEKSHttpAuthSchemeParametersProvider [as httpAuthSchemeParametersProvider]
	/node_modules/@aws-sdk/client-eks/dist-cjs/auth/httpAuthSchemeProvider.js:9
  • index.js	
	/node_modules/@smithy/core/dist-cjs/index.js:61
  • index.js	
	/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:34
  • index.ts	VpcCniAddOn.provideVersion
	/node_modules/@aws-quickstart/eks-blueprints/lib/addons/core-addon/index.ts:187
2024-06-11 19:58:59.399	WARN	/node_modules/@aws-quickstart/eks-blueprints/lib/addons/core-addon/index.ts:209	main	Failed to retrieve add-on versions from EKS for add-on vpc-cni. Falling back to default version.
```

### Reproduction Steps

Install any of the three aforementioned add-ons and run cdk deploy.  The error seems to reliably happen. 

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.133.0

### EKS Blueprints Version

1.14.1

### Node.js Version

v18.20.1

### Environment details (OS name and version, etc.)

MacOS 14.5 Sonoma ARM64

### Other information

_No response_

### PR DESCRIPTION
**Pull request by shapirov103**
despite an explicit setting on the blueprint. Resolves #1015

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: synth.log [hunk 1]
```diff
@@ -0,0 +1,1324 @@
+
+> @aws-quickstart/eks-blueprints@1.15.1 examples
```
Comment (elamaran11) [inline synth.log:2]: What is this file? Why is this added to PR?

**Review by elamaran11 — COMMENTED** [review]
One comment.

File: synth.log (hunk 1 above)
Comment (shapirov103) [reply synth.log]: removed, good catch.

File: test/cloudwatch-insights.test.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import * as blueprints from '../lib';
 import {CloudWatchInsightsAddOnProps, Values} from "../lib";
 import {Match, Template} from "aws-cdk-lib/assertions";
 import {KubernetesVersion} from "aws-cdk-lib/aws-eks";
+import { add } from 'lodash';
```
Comment (elamaran11) [inline test/cloudwatch-insights.test.ts:6]: Remove this line.

**Review by elamaran11 — COMMENTED** [review]
One more minor feedback.

File: test/cloudwatch-insights.test.ts (hunk 2 above)
Comment (shapirov103) [reply test/cloudwatch-insights.test.ts]: addressed

**Review by elamaran11 — APPROVED** [review]
LGTM
